src/main/java/com/cleanroommc/groovyscript/compat/vanilla/Commands.java

```diff
@@ -0,0 +1,99 @@
+package com.cleanroommc.groovyscript.compat.vanilla;
+
+import com.cleanroommc.groovyscript.GroovyScript;
+import com.cleanroommc.groovyscript.api.GroovyBlacklist;
+import com.cleanroommc.groovyscript.api.IScriptReloadable;
+import com.cleanroommc.groovyscript.command.SimpleCommand;
+import com.cleanroommc.groovyscript.core.mixin.CommandHandlerAccessor;
+import com.cleanroommc.groovyscript.registry.AbstractReloadableStorage;
+import com.cleanroommc.groovyscript.registry.NamedRegistry;
+import net.minecraft.command.CommandHandler;
+import net.minecraft.command.ICommand;
+import net.minecraft.server.MinecraftServer;
+import net.minecraftforge.client.ClientCommandHandler;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+
+public class Commands extends NamedRegistry implements IScriptReloadable {
+
+    private final List<ICommand> serverCommands = new ArrayList<>();
+    private final AbstractReloadableStorage<ICommand> serverReloadableCommands = new AbstractReloadableStorage<>();
+    private final AbstractReloadableStorage<ICommand> clientReloadableCommands = new AbstractReloadableStorage<>();
+    private boolean serverStarted = false;
+
+    public void registerCommand(ICommand command) {
+        if (GroovyScript.getSandbox().isRunning() && GroovyScript.getSandbox().getCurrentLoader().isReloadable()) {
```

Comment:
i feel like this code being required for `Commands` to work in preInit (if im reading what the reason is correctly) might indicate a deeper issue.

Reply:
? I just dont want commands to reload when not registered durin postInit. I dont see any issue